app/src/main/java/com/asfoundation/wallet/ui/iab/InAppPurchaseInteractor.java

```diff
@@ -284,4 +292,31 @@ private PaymentMethod mapPaymentMethods(PaymentMethodEntity paymentMethod,
     return new PaymentMethod(paymentMethod.getId(), paymentMethod.getLabel(),
         paymentMethod.getIconUrl(), false);
   }
+
+  boolean hasPreSelectedPaymentMethod() {
+    return sharedPreferences.contains(PRE_SELECTED_PAYMENT_METHOD_KEY);
+  }
+
+  String getPreSelectedPaymentMethod() {
+    return sharedPreferences.getString(PRE_SELECTED_PAYMENT_METHOD_KEY,
+        PaymentMethodsView.PaymentMethodId.APPC_CREDITS.getId());
+  }
+
+  void savePreSelectedPaymentMethod(String paymentMethod) {
+    SharedPreferences.Editor editor = sharedPreferences.edit();
+    editor.putString(PRE_SELECTED_PAYMENT_METHOD_KEY, paymentMethod);
+    editor.putString(LAST_USED_PAYMENT_METHOD_KEY, paymentMethod);
```

Comment:
Isn’t the last always the same as the pre-selected?

Reply:
PRE_SELECTED_PAYMENT_METHOD_KEY is for the new flow where we don't show the payment methods list.
LAST_USED_PAYMENT_METHOD_KEY is used when there is no pre selected payment method but we want to have the radio check on the last used (before clicking on "more payment method)